src/client/BaseClient.js

```diff
@@ -53,8 +61,9 @@ class BaseClient extends EventEmitter {
    * Destroys all assets used by the base client.
    */
   destroy() {
-    for (const t of this._timeouts) clearTimeout(t);
-    for (const i of this._intervals) clearInterval(i);
+    for (const t of this._timeouts) this.clearTimeout(t);
+    for (const i of this._intervals) this.clearInterval(i);
+    for (const i of this._immediates) this.clearImmediate(i);
```

Comment:
Although that this should be empty after this loop, a `this._immediates.clear();` should be at the end of this method for consistency with the other two.